### DISCUSSION TITLE
Combobox with objects - not simple string items

### DISCUSSION THREAD

**Discussion by shurshilov** [mantinedev/mantine#7049] [Feedback]
I'm trying to get the combobox to work as a many2one relation selection field. This is a frequent case in my opinion. And the problem is that I can't figure out how to get him to work not just with string like
const MOCKDATA = [
  '🍎 Apples',
  '🍌 Bananas',
  '🥦 Broccoli',
  '🥕 Carrots',
  '🍫 Chocolate',
  '🍇 Grapes',
];
And with complex strings, the selection of which occurs as a result by ID, but the display by NAME
const MOCKDATA = [
  {3,'Apples'},
  {4,'Bananas'},
  {5,'Broccoli'},
  {6,'Carrots'},
  {7,'Chocolate'},
];
the idea is that eventually the record ID will be in the form. But the user will choose it by its string name
We cannot provide an example?

**Comment by rtivital** [comment]
This logic is implemented in Select component, you can view its source code and use it as a reference for your custom component.

**Reply by shurshilov** [reply]
Tell me the link please, I'm looking at examples of the select there as well, just simple lists of strings
I just don't understand in other design system these are at least 2 values of value and label. Here is an example from antd. Why do you just have one line...
https://codesandbox.io/p/sandbox/basic-usage-antd-5-21-6-msnv5x?file=%2Fdemo.tsx

**Reply by shurshilov** [reply]
> This logic is implemented in Select component, you can view its source code and use it as a reference for your custom component.

Did you mean to insert select inside the combobox instead of inputbase?

**Reply by rtivital** [reply]
You can find the Select component source code here – https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Select/Select.tsx
